src/main/resources/schema/provider.definition.schema.v1.json

```diff
@@ -9,23 +9,46 @@
             "pattern": "^https://[0-9a-zA-Z]([-.\\w]*[0-9a-zA-Z])(:[0-9]*)*([?/#].*)?$",
             "maxLength": 4096
         },
-        "handlerDefinition": {
+        "cudHandlerDefinition": {
             "description": "Defines any execution operations which can be performed on this resource provider",
             "type": "object",
             "properties": {
                 "permissions": {
-                    "type": "array",
-                    "items": {
-                        "type": "string"
-                    },
-                    "additionalItems": false
+                    "$ref": "#/definitions/permissionsDefinition"
+                },
+                "timeoutInMinutes": {
+                    "description": "Defines the timeout for the entire operation to be interpreted by the invoker of the handler.  The default is 120 (2 hours).",
+                    "type": "integer",
+                    "minimum": 2,
+                    "maximum": 720,
+                    "default": 120
```

Comment:
Isn't this for IAM roles and these should range from [1 hour to 12 hours](https://aws.amazon.com/blogs/security/enable-federated-api-access-to-your-aws-resources-for-up-to-12-hours-using-iam-roles/)?

Reply:
720s = 12h? don't see any indication in that link that less than 1h could be a problem?

Reply:
this is not just for IAM roles, this is for the workflow timeout.  Even if the minimum role was an hour the handler could still timeout after 2 minutes.